synapse/lib/ast.py

```diff
@@ -2988,7 +2988,12 @@ async def compute(self, runt, path):
         if self.isconst:
             return self.constval
 
-        vals = [await tostr(await k.compute(runt, path)) for k in self.kids]
+        vals = []
+        for kid in self.kids:
+            part = await kid.compute(runt, path)
+            if part is None:
+                raise s_exc.BadTypeValu(mesg='Null values in tag names are not allowed.')
```

Comment:
Can we know the variable name that resolved to `null` here and include it?

Reply:
Yeah that would be helpful, added